src/amo/components/CategoryHeader/index.js

```diff
@@ -0,0 +1,83 @@
+import classNames from 'classnames';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { compose } from 'redux';
+
+import { ADDON_TYPE_EXTENSION, ADDON_TYPE_THEME } from 'core/constants';
+import translate from 'core/i18n/translate';
+import { getCategoryColor } from 'core/utils';
+import Card from 'ui/components/Card';
+import LoadingText from 'ui/components/LoadingText';
+
+import './styles.scss';
+
+
+export class CategoryHeaderBase extends React.Component {
+  static propTypes = {
+    category: PropTypes.object.isRequired,
+    i18n: PropTypes.object.isRequired,
+  }
+
+  render() {
+    const { category, i18n } = this.props;
+
+    const addonType = category && category.type ? category.type : null;
+    const color = getCategoryColor(category);
+
+    // This is here until
+    // https://github.com/mozilla/addons-server/issues/5728
+    // as well as
+    // https://github.com/mozilla/addons-server/issues/5729 are fixed.
+    // Also, if we don't get a description for a category (maybe it's a new one
+    // or lacks a description) we'll show this generic message.
+    if (category && !category.description) {
+      category.description = i18n.gettext(
+        'Browse all add-ons in this category.');
+
+      if (category.type === ADDON_TYPE_EXTENSION) {
+        category.description = i18n.gettext(
+          'Browse all extensions in this category.');
+      }
+
+      if (category.type === ADDON_TYPE_THEME) {
+        category.description = i18n.gettext(
+          'Browse all themes in this category.');
+      }
+    }
+
+    return (
+      <Card
+        className={classNames('CategoryHeader', {
+          'CategoryHeader--loading': !category,
+          [`CategoryHeader--category-color-${color}`]: color,
+          [`CategoryHeader--type-${addonType}`]: addonType,
+        })}
+      >
+        <div className="CategoryHeader-wrapper">
+          <div className="CategoryHeader-contents">
+            <h1 className="CategoryHeader-name">
+              {category ? category.name : <LoadingText />}
+            </h1>
+            <div className="CategoryHeader-description">
+              {category ? (
+                <p className="CategoryHeader-paragraph">
+                  {category.description}
+                </p>
+              ) : (
+                <p className="CategoryHeader-paragraph">
```

Comment:
We need an element wrapping things in React so that won't work because we have three elements below. It's better to do it this way than put a <div> inside the <p> around the loading texts.